tests/integration/test_check_downloaded_output.py

```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from os import path
+import tarfile
+from pathlib import Path
+
+import utils
+
+
+def test_check_downloaded_output(test_env, create_new_request, tmpdir):
+    """
+    Check that the bundle has all the necessities.
+
+    Process:
+    * Send new request to Cachito API
+    * Send request to download appropriate bundle from Cachito
+
+    Checks:
+    * Check that response code is 200
+    * Check that state is "complete"
+    * Check the downloaded data are in gzip format and valid
+    * Check that dir deps/gomod/… contains cached dependencies
+    * Check that dir app/ contains application source code
+    * Check that the same full path filename is not duplicated
+    """
+    response = create_new_request
+    assert response.status == 201
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response = client.wait_for_complete_request(response)
+    assert response.data["state"] == "complete"
+
+    file_name = tmpdir.join(f"download_{str(response.id)}")
+    file_name_tar = tmpdir.join(f"download_{str(response.id)}.tar.gz")
+
+    resp = client.download_bundle(response.id, file_name_tar)
+    assert resp.status == 200
+    assert tarfile.is_tarfile(file_name_tar)
+
+    with tarfile.open(file_name_tar, "r:gz") as tar:
+        tar.extractall(file_name)
+
+    dependencies_path = "deps/gomod/pkg/mod/cache/download/"
+    names = [i["name"] for i in response.data["dependencies"]]
+    for dependency in names:
+        package_name = utils.escape_path_go(dependency)
+        dependency_path = path.join(file_name, dependencies_path, package_name)
+        assert path.exists(dependency_path)
+
+    assert path.exists(f"{file_name}/app/go.mod")
+    with open(f"{file_name}/app/go.mod", "r") as file:
+        module_name = list()
```

Comment:
When you create variables that are lists, could you make the names plural (e.g. `module_names`)? It makes it easier to read.